libcloud/compute/drivers/docker.py

```diff
@@ -0,0 +1,558 @@
+        if host.startswith('https://'):
+            secure = True
+
+        # strip the prefix
+        prefixes = ['http://', 'https://']
+        for prefix in prefixes:
+            if host.startswith(prefix):
+                host = host.strip(prefix)
+
+        if key_file or cert_file:
+            # docker tls authentication-
+            # https://docs.docker.com/articles/https/
+            # We pass two files, a key_file with the
+            # private key and cert_file with the certificate
+            # libcloud will handle them through LibcloudHTTPSConnection
+            if not (key_file and cert_file):
+                raise Exception(
+                    'Needs both private key file and '
+                    'certificate file for tls authentication')
+            self.connection.key_file = key_file
+            self.connection.cert_file = cert_file
+            self.connection.secure = True
+        else:
+            self.connection.secure = secure
+
+        self.connection.host = host
+        self.connection.port = port
+
+    def create_node(self, name, image, command=None, hostname=None, user='',
+                    stdin_open=True, tty=True,
+                    mem_limit=0, ports=None, environment=None, dns=None,
+                    volumes=None, volumes_from=None,
+                    network_disabled=False, entrypoint=None,
+                    cpu_shares=None, working_dir='', domainname=None,
+                    memswap_limit=0, port_bindings=None):
+        """
+        Create a container
+
+        Create a container, based on an image and optionally specify command
+        and other settings. If image is not found, try to pull it
+        After the container is created, start it
+        """
+        command = shlex.split(str(command))
+        if port_bindings is None:
+            port_bindings = {}
+        params = {
+            'name': name
+        }
+
+        payload = {
+            'Hostname': hostname,
+            'Domainname': domainname,
+            'ExposedPorts': ports,
+            'User': user,
+            'Tty': tty,
+            'OpenStdin': stdin_open,
+            'StdinOnce': False,
+            'Memory': mem_limit,
+            'AttachStdin': True,
+            'AttachStdout': True,
+            'AttachStderr': True,
+            'Env': environment,
+            'Cmd': command,
+            'Dns': dns,
+            'Image': image,
+            'Volumes': volumes,
+            'VolumesFrom': volumes_from,
+            'NetworkDisabled': network_disabled,
+            'Entrypoint': entrypoint,
+            'CpuShares': cpu_shares,
+            'WorkingDir': working_dir,
+            'MemorySwap': memswap_limit,
+            'PublishAllPorts': True,
+            'PortBindings': port_bindings,
+        }
+
+        data = json.dumps(payload)
+        try:
+            result = self.connection.request('/containers/create', data=data,
```

Comment:
Makes sense! :+1: 

As I said, I asked for curiosity, and to understand your development choice! :)